sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/vaccination/VaccinationEditActivity.java

```diff
@@ -0,0 +1,181 @@
+        return DatabaseHelper.getVaccinationDao().queryUuid(recordUuid);
+    }
+
+    @Override
+    protected VaccinationEntity buildRootEntity() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean result = super.onCreateOptionsMenu(menu);
+        getSaveMenu().setTitle(R.string.action_save_vaccination);
+        return result;
+    }
+
+    @Override
+    public List<PageMenuItem> getPageMenuData() {
+        List<PageMenuItem> menuItems = PageMenuItem.fromEnum(VaccinationSection.values(), getContext());
+        return menuItems;
+    }
+
+    @Override
+    protected BaseEditFragment buildEditFragment(PageMenuItem menuItem, VaccinationEntity activityRootData) {
+        VaccinationSection section = VaccinationSection.fromOrdinal(menuItem.getPosition());
+        BaseEditFragment fragment;
+        switch (section) {
+
+            case VACCINATION_INFO:
+                fragment = VaccinationEditFragment.newInstance(activityRootData);
+                break;
+            case HEALTH_CONDITIONS:
+                fragment = VaccinationEditHealthConditionsFragment.newInstance(activityRootData);
+                break;
+            default:
+                throw new IndexOutOfBoundsException(DataHelper.toStringNullable(section));
+        }
+
+        return fragment;
+
+    }
+
+    @Override
+    protected int getActivityTitle() {
+        return R.string.heading_vaccination_edit;
+    }
+
+    @Override
+    public void saveData() {
+        if (saveTask != null) {
+            NotificationHelper.showNotification(this, WARNING, getString(R.string.message_already_saving));
+            return;
+        }
+
+        final VaccinationEntity vaccinationEntity = getStoredRootEntity();
+
+        try {
+            FragmentValidator.validate(getContext(), getActiveFragment().getContentBinding());
+        } catch (ValidationException e) {
+            NotificationHelper.showNotification(this, ERROR, e.getMessage());
+            return;
+        }
+
+        saveTask = new SavingAsyncTask(getRootView(), vaccinationEntity) {
+
+            @Override
+            protected void onPreExecute() {
+                showPreloader();
+            }
+
+            @Override
+            public void doInBackground(TaskResultHolder resultHolder) throws DaoException {
+                final VaccinationEntity savedVaccination = DatabaseHelper.getVaccinationDao().saveAndSnapshot(vaccinationEntity);
+                final Immunization immunization = DatabaseHelper.getImmunizationDao().queryUuid(vaccinationEntity.getImmunization().getUuid());
+                final List<VaccinationEntity> vaccinations = immunization.getVaccinations();
+                if (vaccinationEntity.getId() == null) {
+                    vaccinations.add(savedVaccination);
+                } else {
+                    VaccinationEntity vaccinationToReplace = null;
```

Comment:
You don't have to create a variable for this purpose, simply move the `remove` call to inside the for loop and then add a break.